### DISCUSSION TITLE
[@types/ssh2] claims Session extends Duplex but it only extends EventEmitter

### DISCUSSION THREAD

**Discussion by slifty** [DefinitelyTyped/DefinitelyTyped#64281] [Issues with a @types package]
This issue recently resulted in some unexpected errors / bypasses type safety so I wanted to raise it in case there is a solution.

According to https://github.com/DefinitelyTyped/DefinitelyTyped/blob/29e78e018dc6c7d49f7bb101064da9163f7f8cf5/types/ssh2/index.d.ts#L1112 this is already a known issue, and it was introduced / changed as part of this PR: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/a9706fdcdf619bf5a43b7b5bd0e8388ca25fdbf1 by @dhensby 

Would it be possible to update the `Session` type to reflect its actual properties or is there something deeper going on?

[a comment by an automated account is not shown]

**Comment by dhensby** [comment]
Sorry for the delay responding here. It was a while ago that I updated these typings. IIRC the main reason I implemented the typings this way was because the intent of the base library seems to be that they should share the same interface and the interface did appear to be correctly poly-filled.

Do you have examples of where this is causing issues?
